compiler/quilt/nodes.py

```diff
@@ -47,20 +47,30 @@ def __init__(self, package, node, data, meta):
         self._node = node
         self.__cached_data = data
 
-    def _data(self):
+    def _data(self, asa=None):
         """
-        Returns the contents of the node: a dataframe or a file path.
+        Returns the contents of the node: a dataframe or a file path, or passes
+        the node and is contents to a callable.        
         """
-        if self.__cached_data is None:
-            # TODO(dima): Temporary code.
+        if callable(asa):
             store = self._package.get_store()
```

Comment:
What do we do if the node is created using `_set(...)`, and doesn't have a hash?

Reply:
The only things that make sense to me are: (1) return cached data, (2) pass cached data to asa, or (3) assert. What's your preference? We could also add a parameter to the asa signature to accepted cached data.

Reply:
Well... I think 1) would be incorrect because that's equivalent to just ignoring `asa`; 2) would likely break because the `asa` function normally accepts file paths, not dataframes. I guess 3) then? Though a `ValueError` or something similar is better than `assert` since it's not a programming error.